### PR TITLE
Add Citra AppImage builds

### LINKED ISSUE #2665: Please provide an AppImage for Linux
Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- Works for most Linux distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others)
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed
- Optional desktop integration with `appimaged`
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can optionally GPG2-sign your AppImages (inside the file)
- Works on Live ISOs
- Can use the same AppImages when dual-booting multiple distributions

The [linuxdeployqt](https://github.com/probonopd/linuxdeployqt) tool makes it easy to bundle Qt-based applications in AppImage format.

[Here is an overview](https://github.com/probonopd/AppImageKit/wiki/AppImages) of projects that are already distributing upstream-provided, official AppImages.

If you have questions, AppImage developers are on #AppImage on irc.freenode.net.

Reference: https://www.reddit.com/r/Citra/comments/63y6zv/linux_mint_user_here_what_to_do_about_glibcxx/

### PR DESCRIPTION
**Pull request by TGP17**
This PR adds Citra AppImage builds. This PR modifies .github/workflows/ci.yml to add AppImage builds and also adds linux-appimage folder in the .ci folder, which contains the necessary  scripts to build the AppImage. This fixes #2665

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6404)

### CODE REVIEW

File: .ci/linux-appimage/install_dependencies.sh
Comment (SachinVin) [inline .ci/linux-appimage/install_dependencies.sh:1]: Use a docker container to cache dependencies

File: .ci/linux-appimage/build.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+mv ./AppDir/usr/local/include ./AppDir/usr
+mv ./AppDir/usr/local/lib ./AppDir/usr
+mv ./AppDir/usr/local/share ./AppDir/usr
+rm -rf ./AppDir/usr/local
+
+#Downloading Tools for Building AppImages
+wget https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage
+wget https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous/linuxdeploy-plugin-qt-x86_64.AppImage
+chmod a+x linuxdeploy-x86_64.AppImage
+chmod a+x linuxdeploy-plugin-qt-x86_64.AppImage
```
Comment (SachinVin) [inline .ci/linux-appimage/build.sh:23]: Should be fetched with other dependencies

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -55,6 +55,23 @@ jobs:
...
           name: ${{ matrix.image }}
           path: artifacts/
+  build_linux-appimage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Make scripts executable
+        run: chmod a+x ./.ci/linux-appimage/install_dependencies.sh && chmod a+x ./.ci/linux-appimage/build.sh
```
Comment (SachinVin) [inline .github/workflows/ci.yml:65]: Update permission on git instead of doing it here

File: .ci/linux-appimage/build.sh (hunk 1 above)
Comment (MT19048) [reply .ci/linux-appimage/build.sh:23]: I did this similar to Yuzu. They fetch their AppImage scripts in the build script. But I'll try to include it in the dockerfile I'm currently working on

**Comment by qurious-pixel** [comment]
Use ubuntu 20.04 to lower the glibc version in the Docker.

File: .ci/linux-appimage/build.sh (hunk 1 above)
Comment (SachinVin) [reply .ci/linux-appimage/build.sh:23]: @liushuyu wouldn't it be fine for these to be in docker, or do we need to fetch the latest version from the build script ?

File: .ci/linux-appimage/install_dependencies.sh
Comment (MT19048) [reply .ci/linux-appimage/install_dependencies.sh:1]: Alright, I made a PR on citra-emu/build-environments to add my dockerfile

File: .ci/linux-appimage/build.sh (hunk 1 above)
Comment (MT19048) [reply .ci/linux-appimage/build.sh:23]: I added them to my dockerfile, but I can't seem to figure out how to access that from the build script.

**Comment by qurious-pixel** [comment]
`post-upload.sh` and `upload.sh` are doing similiar functions so that it creates two types of compressions `tar.xz` and `7z` packaged into one zip (with two copies of the same files).
 
Place everything into `upload.sh` and compress only once.

**Comment by SachinVin** [comment]
I think it is fine to directly copy the `.AppImage` to the artifacts folder, you will have to update [this line](https://github.com/citra-emu/citra/blob/master/.github/workflows/ci.yml#L237), so releases are properly updated

**Comment by MT19048** [comment]
> I think it is fine to directly copy the `.AppImage` to the artifacts folder, you will have to update [this line](https://github.com/citra-emu/citra/blob/master/.github/workflows/ci.yml#L237), so releases are properly updated

Alrighty, done

File: .ci/linux-appimage/upload.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -ex
+
+. .ci/common/pre-upload.sh
+
+mv build/Citra*.AppImage "${GITHUB_WORKSPACE}"/artifacts/
```
Comment (SachinVin) [inline .ci/linux-appimage/upload.sh:5]:
```suggestion
REV_NAME="citra-linux-${GITDATE}-${GITREV}"
mv build/Citra*.AppImage "${GITHUB_WORKSPACE}"/artifacts/${REV_NAME}.AppImage
```

Comment (TGP17) [reply .ci/linux-appimage/upload.sh]: Alright I updated Upload.sh to include it

Comment (SachinVin) [reply .ci/linux-appimage/upload.sh]: Seems you didn't change to `mv build/Citra*.AppImage "${GITHUB_WORKSPACE}"/artifacts/${REV_NAME}.AppImage` (appending `/${REV_NAME}.AppImage`)

Comment (TGP17) [reply .ci/linux-appimage/upload.sh]: You're right sorry, I somehow missed that. I added that now too

**Review by Steveice10 — APPROVED** [review]

**Pull request merged**